src/better-fetch.js

```diff
@@ -0,0 +1,41 @@
+class FetchError extends Error {
```

Comment:
This is overkill! Just add x amount of retries and a sleep, maybe make it an exponential backoff.
https://en.wikipedia.org/wiki/Exponential_backoff